ibis/omniscidb/client.py

```diff
@@ -488,6 +488,51 @@ def alter(self, tbl_properties=None):
         """
         raise NotImplementedError('This method is not implemented yet!')
 
+    def insert(
+        self,
+        obj: Union[pd.DataFrame, ir.Expr],
+        dst_cols: Optional[list] = None,
```

Comment:
why is this argument needed?

Reply:
OmniSci has two ways to insert data into table:
1) INSERT INTO <dst_table> VALUES (<value>, ...);
2) INSERT INTO <dst_table> (<column>, ...) VALUES (<value>, ...);

or for select

1) INSERT INTO <dst_table> SELECT * FROM <source_table>;
2) INSERT INTO <dst_table> (<column>, ...) SELECT * FROM <source_table>;

Thus, user may want to specify column names. That's why I added the corresponding parameter.

Reply:
hey @YarShev ! 
thanks for working on that. I was reviewing the impala implementation and it seems that there is no support for `values`. 

not sure if `values` and `dst_cols` would be something used frequently. 

I took it from the tests:

```
    values = [
        'polygon((0 0, 4 0, 4 4, 0 4, 0 0), (1 1, 2 1, 2 2, 1 2, 1 1))',
        'point(1 1)',
        1,
        2.0,
    ]
    dst_cols = ['a', 'b', 'c', 'd']
```
I think it is better represented using just pandas or geopandas.

Is there any use case that pandas or geopandas doesn't work for that?

about the `obj` implementation looks very nice! I will review the entire PR today.

I am curious about other reviewers think about it. cc @jreback @datapythonista

Reply:
Hey @xmnlab , thanks a lot for your review!
I think we can keep `values` and `dst_cols` parameters because they may be used in some cases (even though these use cases is no so much).
Regarding to `values` and `dst_cols` which were taken from tests by you. About 1-2 months ago I tested with both geopandas and string values and didn't find any issues. Here I wanted to show how can these values can be passed using string format to do not add additional imports. Thus, I don't see any bad thinks if these parameters are passed through string format.
Since this PR is on review for long time I'd like merge it as soon as possible to do not return to it and move next. 
From my point of view, all comments which were here from you and @jreback are addressed and implementation works correctly.
@jreback , review too, please. @xmnlab , one more time, thanks.

Reply:
Didn't check the implementations of insert in other backends, but this signature is overcomplicated.

For now better remove `dst_cols`, you can propose adding it in a follow up PR.

Then, if I understood correctly, you can insert by providing the values in a pandas DataFrame in `obj`, or in a list in `values`. If that's correct, I think it's better to provide a single value `values`, that can be both, a pandas DataFrame, or a list.

For the list, please specify the type of the list in the type annotations.

For `validate`, why can the parameter be optional? Why is `None` different than `False`?